File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/LibraryCardForm.ts
import { LookupEditor, ServiceLookupEditor, EnumEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { LibraryCardStatusEnum } from "../Modules/ReaderManage.LibraryCard.LibraryCardStatusEnum";

export interface LibraryCardForm {
    UserId: LookupEditor;
    LevelId: ServiceLookupEditor;
    Status: EnumEditor;
}

export class LibraryCardForm extends PrefixedContext {
    static readonly formKey = 'ReaderManage.LibraryCard';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!LibraryCardForm.init)  {
            LibraryCardForm.init = true;

            var w0 = LookupEditor;
            var w1 = ServiceLookupEditor;
            var w2 = EnumEditor;

            initFormType(LibraryCardForm, [
                'UserId', w0,
                'LevelId', w1,
                'Status', w2
            ]);
        }
    }
}

[LibraryCardStatusEnum]; // referenced types

File: LMIS/LMIS.Web/Modules/ReaderManage/LibraryCard/LibraryCardForm.cs
using LMIS.Modules.ReaderManage.LibraryCard;
using Serenity.ComponentModel;
using System;

namespace LMIS.ReaderManage.Forms;

[FormScript("ReaderManage.LibraryCard")]
[BasedOnRow(typeof(LibraryCardRow), CheckNames = true)]
public class LibraryCardForm
{
    public long UserId { get; set; }
    public long LevelId { get; set; }
    public LibraryCardStatusEnum Status { get; set; }
}